blockchain/blockchain.go

```diff
@@ -678,6 +681,9 @@ func (bc *blockchain) MintNewSecretBlock(
 func (bc *blockchain) CommitBlock(blk *Block) error {
 	bc.mu.Lock()
 	defer bc.mu.Unlock()
+	if b, _ := bc.GetBlockByHash(blk.HashBlock()); b != nil {
```

Comment:
This may not work perfectly. Between L686 and L687, if there's a context switch, the db will still throw error. I don't think we need to define another error. Instead, if we see the cause of the error is `db.ErrAlreadyExist`, we should just ignore it, and not print error message.